UI/obs-frontend-api/obs-frontend-api.h

```diff
@@ -208,7 +208,9 @@ EXPORT obs_output_t *obs_frontend_get_recording_output(void);
 EXPORT obs_output_t *obs_frontend_get_replay_buffer_output(void);
 
 EXPORT config_t *obs_frontend_get_profile_config(void);
-EXPORT config_t *obs_frontend_get_global_config(void);
+EXPORT config_t *obs_frontend_get_global_config(void) OBS_DEPRECATED;
+EXPORT config_t *obs_frontend_get_app_config(void);
+EXPORT config_t *obs_frontend_get_user_config(void);
```

Comment:
This is something I wish we stop doing.
I think we should create another way for add-ons to put their settings and never have to touch a `config_t *`.

Reply:
I agree, but the PR is massive enough as it is and I tried to focus on one thing without rocking the boat too much.

E.g. with these changes it should be easier to encapsulate even further and implement something like `OBSConfig` or `OBSProfile` classes which are interacted with (so instead of `GetCurrentProfile` returning a struct, it returns a reference to a class instance) so instead of taking the profile's path and adding `recordEncoder.json` to it, you would do `OBSProfile::GetRecordEncoder()` or something.

But - again - would have been out of scope, but is definitely where we should want to end up.

Reply:
In my opinion adding new API is not a middle-ground, it just "worsen" to at least 2-3 major version.

Reply:
Just removing access to the configuration objects without offering an alternative is not an option either.

And - again - the defined scope for this change was to just make the locations of these things to be adjustable, not to change the way plugins interact with the settings system.

Such a change can be done, but would constitute a rewrite of large chunks of the application and would be stuck in review-hell for eternity.

Reply:
Feel free to mark it as resolve, I have no motivation.